src/main/java/Duke.java

```diff
@@ -1,10 +1,172 @@
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
+        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+
+        userInput.setPrefWidth(325.0);
+
+        sendButton.setPrefWidth(55.0);
+
+        AnchorPane.setTopAnchor(scrollPane, 1.0);
+
+        AnchorPane.setBottomAnchor(sendButton, 1.0);
+        AnchorPane.setRightAnchor(sendButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput, 1.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+
+        stage.setScene(scene);
+        stage.show();
+
+        //Scroll down to the end every time dialogContainer's height changes.
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    /**
+     * Instantiates a new Duke object.
+     * Load an existing .txt file for the list of tasks stored.
+     *
+     * @param filePath The file path of the list .txt file
+     */
+    public Duke(String filePath) {
+        try {
+            storage = new Storage(filePath);
+            tasks = new TaskList(storage.loadFile()); // add existing tasks from txt file onto ArrayList
+            size = tasks.getSize(); // get existing size of list
+            storage.closeFile();
+        } catch (DukeException e) {
+            ui.loadError();
+            ui.createFile();
+            ArrayList<Task> list = new ArrayList<>();
+            tasks = new TaskList(list);
+            File file = new File(filePath); // create new .txt file with provided filepath
+            try {
+                file.createNewFile();
+            } catch (Exception err) {
+                System.out.println("Unable to create new file."); // Unlikely
+            }
+        }
+    }
+
+    /**
+     * Run the Duke application.
+     * Accepted commands: list, done, delete, find
+     * Accepted tasks: todo, event, deadline
+     */
+    public void run() {
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        // Store command-line input as String
+        String input = "";
+
+        Storage storage = this.storage;
+        TaskList tasks = this.tasks;
+        Parser parser = new Parser();
+
+        // Create new user interface object
+        Ui ui = new Ui();
+
+        // Initial opening introduction and prompt for user input
+        ui.openingStatement();
+
+        // Parse Duke commands
+        parser.parseDuke(storage, tasks, input);
```

Comment:
Is there a reason why you didn't create a separate Command Class? Would separating the parsing and execution of commands make more sense?